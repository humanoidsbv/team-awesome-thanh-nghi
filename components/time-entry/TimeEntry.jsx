import React from 'react';
import PropTypes from 'prop-types';

import './time-entry.scss';

class TimeEntry extends React.Component {
  confirmDelete = (id) => {
    const { onDelete } = this.props;
    const askConfirm = window.confirm('Do you want to delete this entry?');
    if (askConfirm) {
      onDelete(id);
    }
  }

  render() {
    const {
      client, endTime, id, startTime
    } = this.props;

    return (
      <React.Fragment>
        <ul className="entry-content">
          <li className="entry-content__activity">
            <div className="entry-content__client">{client}</div>
            <button
              onClick={() => this.confirmDelete(id)}
              className="entry-content__button"
              type="button"
            >
              <img
                alt="Delete icon"
                className="entry-content__button-icon"
                src="/static/images/icon-delete.svg"
              />
              Delete
            </button>
            <div className="entry-content__time">
              {`${startTime} - ${endTime}`}
            </div>
          </li>
        </ul>
      </React.Fragment>
    );
  }
}

TimeEntry.propTypes = {
  client: PropTypes.string.isRequired,
  endTime: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  onDelete: PropTypes.func.isRequired,
  startTime: PropTypes.string.isRequired
};

export default TimeEntry;
