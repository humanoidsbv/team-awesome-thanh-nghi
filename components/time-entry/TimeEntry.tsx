import React from 'react';

import './time-entry.scss';
import { TimeEntriesState } from 'ducks/time-entries';

interface TimeEntryProps {
  client: string;
  endTime: string;
  id: string;
  onDelete: Function;
  startTime: string;
}

class TimeEntry extends React.Component<TimeEntryProps, TimeEntriesState> {

  confirmDelete = (id) => {
    const { onDelete } = this.props;
    const askConfirm = window.confirm('Do you want to delete this entry?');
    if (askConfirm) {
      onDelete(id);
    }
  }

  render() {
    const {
      client, endTime, id, startTime
    } = this.props;

    return (
      <React.Fragment>
        <ul className="entry-content">
          <li className="entry-content__activity">
            <div className="entry-content__client">{client}</div>
            <button
              onClick={() => this.confirmDelete(id)}
              className="entry-content__button"
              type="button"
            >
              <img
                alt="Delete icon"
                className="entry-content__button-icon"
                src="/static/images/icon-delete.svg"
              />
              Delete
            </button>
            <div className="entry-content__time">
              {`${startTime} - ${endTime}`}
            </div>
          </li>
        </ul>
      </React.Fragment>
    );
  }
}

export default TimeEntry;
