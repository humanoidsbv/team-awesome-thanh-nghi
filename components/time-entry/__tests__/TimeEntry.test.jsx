import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import TimeEntry from '../TimeEntry.tsx';

const deleteTimeEntry = () => {};

const timeEntries = {
  client: '3',
  endTime: '2019-03-20T16:00:00.000Z',
  id: '123',
  startTime: '2019-03-20T08:00:00.000Z'
};

it('renders a snapshot', () => {
  const wrapper = shallow(
    <TimeEntry
      client={timeEntries.clientName}
      endTime={timeEntries.endTime}
      id={timeEntries.id}
      onDelete={deleteTimeEntry}
      startTime={timeEntries.startTime}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
