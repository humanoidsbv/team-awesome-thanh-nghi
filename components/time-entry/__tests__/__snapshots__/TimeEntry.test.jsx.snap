// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<Fragment>
  <ul
    className="entry-content"
  >
    <li
      className="entry-content__activity"
    >
      <div
        className="entry-content__client"
      />
      <button
        className="entry-content__button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Delete icon"
          className="entry-content__button-icon"
          src="/static/images/icon-delete.svg"
        />
        Delete
      </button>
      <div
        className="entry-content__time"
      >
        2019-03-20T08:00:00.000Z - 2019-03-20T16:00:00.000Z
      </div>
    </li>
  </ul>
</Fragment>
`;
